packages/nodes-base/nodes/Twitter/V2/GenericFunctions.ts

```diff
@@ -84,11 +84,8 @@ export function returnId(tweetId: INodeParameterResourceLocator) {
 		return tweetId.value as string;
 	} else if (tweetId.mode === 'url') {
 		const value = tweetId.value as string;
-		const tweetIdMatch = value.includes('lists')
-			? value.match(/^https?:\/\/twitter\.com\/(?:#!\/)?(\w+)\/list(s)?\/(\d+)$/)
-			: value.match(/^https?:\/\/twitter\.com\/(?:#!\/)?(\w+)\/status(es)?\/(\d+)$/);
-
-		return tweetIdMatch?.[3] as string;
+		const urlParts = value.split('/');
+		return urlParts[urlParts.length - 1];
```

Comment:
This code does not account for query-params.

also, since `value` is expected to be a URL here, we **could** use `new URL` to parse the string to
1. validate that the hostname is either twitter.com or x.com (otherwise throw an error)
2. use extract the tweetId only from `url.pathname` and not from any other part of the url.

Reply:
```suggestion
		try {
			const url = new URL(tweetId.value as string);
			if (/{twitter|x}.com$/.test(url.hostname)) {
				throw new ApplicationError('Invalid domain');
			}
			const parts = url.pathname.split('/');
			if (parts.length < 4 || parts[1] !== 'status') {
				throw new ApplicationError('Invalid path');
			}
			return parts[3];
		} catch (error) {
			throw new ApplicationError('Not a valid tweet url', { level: 'warning', cause: error });
		}
```